### PR TITLE
Irreducible mass

### PR DESCRIPTION
**Pull request by jennifersanchez**
### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/ApparentHorizons/StrahlkorperGr.hpp [hunk 1]
```diff
@@ -295,4 +295,16 @@ std::array<double, 3> spin_vector(double spin_magnitude,
                                   const Scalar<DataVector>& ricci_scalar,
                                   const Scalar<DataVector>& spin_function,
                                   const YlmSpherepack& ylm) noexcept;
+
+/*!
+ * \ingroup SurfacesGroup
+ * \brief Irreducible mass of a 2D `Strahlkorper`.
+ *
+ * \details See Eqs. (15.38)
+ * of Hartle page 325. This function computes the irreducible
```
Comment (jennifersanchez) [inline src/ApparentHorizons/StrahlkorperGr.hpp]: git rid of page number and add title

File: src/ApparentHorizons/StrahlkorperGr.hpp [hunk 2]
```diff
@@ -295,4 +295,16 @@ std::array<double, 3> spin_vector(double spin_magnitude,
...
+ * mass from the area of a horizon 
+ * Specifically, computes
+ * \f$M_{irr}=\sqrt{\frac{A}{16\pi}\f$
```
Comment (jennifersanchez) [inline src/ApparentHorizons/StrahlkorperGr.hpp]: add closed brace for sqrt

Comment (jennifersanchez) [inline src/ApparentHorizons/StrahlkorperGr.hpp]: M_\mathrm{irr}

File: src/ApparentHorizons/StrahlkorperGr.hpp [hunk 3]
```diff
@@ -295,4 +295,16 @@ std::array<double, 3> spin_vector(double spin_magnitude,
...
                                   const Scalar<DataVector>& spin_function,
                                   const YlmSpherepack& ylm) noexcept;
+
+/*!
+ * \ingroup SurfacesGroup
+ * \brief Irreducible mass of a 2D `Strahlkorper`.
+ *
+ * \details See Eqs. (15.38)
+ * of Hartle page 325. This function computes the irreducible
+ * mass from the area of a horizon
```
Comment (jennifersanchez) [inline src/ApparentHorizons/StrahlkorperGr.hpp]: add period

File: tests/Unit/ApparentHorizons/Test_StrahlkorperGr.cpp [hunk 4]
```diff
@@ -279,6 +279,10 @@ void test_area(const Solution& solution, const Strahlkorper<Fr>& strahlkorper,
       strahlkorper.ylm_spherepack().definite_integral(get(area_element).data());
 
   CHECK_ITERABLE_APPROX(area, expected);
+  
+  const double irreducible_mass = StrahlkorperGr::irreducible_mass(area);
+  const double expected_irreducible_mass = sqrt(area/(16.0*M_PI));
+  CHECK_ITERABLE_APPROX(irreducible_mass, expected_irreducible_mass);
```
Comment (jennifersanchez) [inline tests/Unit/ApparentHorizons/Test_StrahlkorperGr.cpp]: `CHECK(irreducible_mass== expected_irreducible_mass)`

File: tests/Unit/ApparentHorizons/Test_StrahlkorperGr.cpp [hunk 5]
```diff
@@ -279,6 +279,10 @@ void test_area(const Solution& solution, const Strahlkorper<Fr>& strahlkorper,
       strahlkorper.ylm_spherepack().definite_integral(get(area_element).data());
 
   CHECK_ITERABLE_APPROX(area, expected);
+  
+  const double irreducible_mass = StrahlkorperGr::irreducible_mass(area);
+  const double expected_irreducible_mass = sqrt(area/(16.0*M_PI));
+  CHECK(irreducible_mass== expected_irreducible_mass);
```
Comment (geoffrey4444) [inline tests/Unit/ApparentHorizons/Test_StrahlkorperGr.cpp]: small thing, but space before ==. I would have thought clang-format would have gotten that for you.

**Review by geoffrey4444 — CHANGES_REQUESTED** [review]
Please run `git-clang-format -f` on your changed *pp files (StrahlkorperGr.cpp, StrahlkorperGr.hpp, Test_StrahlkorperGr.cpp), and fix the minor issue. Then I think this is ready for a second reviewer to look at besides me! Once you make those changes, I'll remove the "in progress" tag and ask for someone else to review this.

**Comment by geoffrey4444** [comment]
Ping @jennifersanchez @GitSamm @denyzamelchor @trami18 ... do you think you can take a stab at my latest change requests today or tomorrow?

**Comment by jennifersanchez** [comment]
Yes. We’ll work on that today.

**Pull request closed without merging**